### PR TITLE
Switch to Airlift and add Controller server

### PR DESCRIPTION
**Pull request by jirassimok**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -62,22 +60,22 @@
             <version>${jackson.version}</version>
         </dependency>
 
-        <dependency>
+        <dependency> <!-- TODO: Remove GSON in favor of Jackson -->
```
Comment (sudarsanm94) [inline pom.xml]: Is it necessary to remove `GSON`?

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/APIFileHandler.java [hunk 2]
```diff
@@ -43,15 +42,15 @@ public Response getFileNameList()
         try {
             List<String> fileNames = AgentFileUtils.getFileNameList(baseDir);
             String namesToReturn = String.join("\r\n", fileNames);
-            LOGGER.debug("Successfully retrieved the list of file names from directory {}", baseDir.toString());
+            LOGGER.debug("Successfully retrieved the list of file names from directory '%s'", baseDir.toString());
             return Response.status(Status.OK).entity(namesToReturn).build();
         }
         catch (IllegalArgumentException e) {
-            LOGGER.error("{} Not a directory", baseDir.toString(), e);
+            LOGGER.error(e, "Pre-configured directory '%s' not a directory", baseDir.toString());
```
Comment (sudarsanm94) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/APIFileHandler.java]: `is` not a directory

File: pom.xml (hunk 1 above)
Comment (ShanBai6) [inline pom.xml]: This happens when you refactor your use of it.

File: pom.xml [hunk 3]
```diff
@@ -62,22 +60,22 @@
...
+            <version>2.1-m09</version>
         </dependency>
 
         <dependency>
-            <groupId>org.apache.logging.log4j</groupId>
-            <artifactId>log4j-core</artifactId>
-            <version>${log4j.version}</version>
+            <groupId>javax.inject</groupId>
+            <artifactId>javax.inject</artifactId>
+            <version>1</version>
```
Comment (ShanBai6) [inline pom.xml]: You already have inject from Guice. Is this necessary?

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentServer.java [hunk 4]
```diff
@@ -13,49 +13,47 @@
...
+                new JsonModule(),
+                new JaxrsModule(true), // requireExplicitBindings = true
+                new HttpEventModule(),
+                new AgentServerModule()
+        );
 
         try {
-            server.start();
-            server.join();
+            Injector injector = bootstrap.initialize();
```
Comment (ShanBai6) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentServer.java]: Maybe add that restrict strict config binding back?

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentServer.java [hunk 5]
```diff
@@ -13,49 +13,47 @@
...
-        }
-        finally {
-            try {
-                server.stop();
-            }
-            catch (Exception e) {
-                // TODO: Log exception
-                throw e;
-            }
+            LOG.error(e, "Error running application");
```
Comment (ShanBai6) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentServer.java]: Should be Error running Agent Server

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentServer.java [hunk 6]
```diff
@@ -13,49 +13,47 @@
...
+import io.airlift.discovery.client.Announcer;
+import io.airlift.discovery.client.DiscoveryModule;
+import io.airlift.event.client.HttpEventModule;
+import io.airlift.http.server.HttpServerModule;
+import io.airlift.jaxrs.JaxrsModule;
+import io.airlift.json.JsonModule;
+import io.airlift.log.Logger;
+import io.airlift.node.NodeModule;
 
 class AgentServer
```
Comment (ShanBai6) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentServer.java]: Consider implements Runnable as Presto does it.

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/CommandExecutor.java [hunk 7]
```diff
@@ -64,7 +63,7 @@ private int execute()
             try (InputStream processStream = process.getInputStream()) {
                 new Thread(() -> {
                     try {
-                        LOGGER.info("Output from command: {}\n{}", commandString, new String(ByteStreams.toByteArray(processStream)));
+                        LOGGER.info("Output from command: %s\n%s", commandString, new String(ByteStreams.toByteArray(processStream)));
                     }
                     catch (IOException e) {
                         LOGGER.error("Failed to log the process output");
```
Comment (ShanBai6) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/CommandExecutor.java]: If we print stack trace once, then we should probably do it all the time. Add e to all error/warning logs.

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/LogsHandler.java [hunk 8]
```diff
@@ -31,6 +30,7 @@
 import java.time.format.DateTimeFormatterBuilder;
 import java.time.format.DateTimeParseException;
 import java.util.function.Predicate;
+import java.util.logging.Level;
```
Comment (ShanBai6) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/LogsHandler.java]: Check package `io.airlift.log` Level enum. We may want to use that

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoUninstaller.java [hunk 9]
```diff
@@ -59,14 +58,14 @@ private static void uninstallUsingRpm(String packageName, boolean checkDependenc
...
-            LOGGER.warn("Package {} is not installed; Process exited with return value: {}", packageName, checkPackageInstalled);
+            LOGGER.warn("Package '%s' is not installed; Process exited with return value: %s", packageName, checkPackageInstalled);
             return;
         }
         if (executeCommand("service", "presto", "status") == 0) {
             if (!ignoreErrors) {
                 throw new PrestoManagerException("Presto is running");
             }
-            LOGGER.warn("Presto is running");
+            LOGGER.warn("Attempted to uninstall, but Presto is running");
```
Comment (sudarsanm94) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoUninstaller.java]: At this stage the uninstall process will continue to happen even if Presto is running. I think the error message should be `Attempting to uninstall despite Presto running` or something like that.

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/api/ConfigAPI.java [hunk 10]
```diff
@@ -43,8 +41,7 @@
 @Singleton
 public final class ConfigAPI
 {
-    private static final APIFileHandler apiFileHandler = new APIFileHandler(Paths.get("/etc/presto"));
-    private static final Logger LOGGER = LogManager.getLogger(ConfigAPI.class);
+    private static final APIFileHandler apiFileHandler = new APIFileHandler(Paths.get("../etc/presto"));
```
Comment (sudarsanm94) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/api/ConfigAPI.java]: For rpm installation the path should be ```/etc/presto```

File: presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/ControllerServer.java [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+import io.airlift.bootstrap.Bootstrap;
+import io.airlift.discovery.client.DiscoveryModule;
+import io.airlift.event.client.HttpEventModule;
+import io.airlift.http.server.HttpServerModule;
+import io.airlift.jaxrs.JaxrsModule;
+import io.airlift.json.JsonModule;
+import io.airlift.log.Logger;
+import io.airlift.node.NodeModule;
+
+class ControllerServer
```
Comment (ShanBai6) [inline presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/ControllerServer.java]: Ditto as Agent Server

File: presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/ControllerServerModule.java [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+import static io.airlift.jaxrs.JaxrsBinder.jaxrsBinder;
+
+public class ControllerServerModule
+        extends AbstractConfigurationAwareModule
+{
+    @Override
+    protected void setup(Binder binder)
+    {
+        binder.disableCircularProxies();
+
```
Comment (ShanBai6) [inline presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/ControllerServerModule.java]: Could we possibly make the JerseyClient as Singleton so that it would only be created once?

File: presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/NodeSet.java [hunk 13]
```diff
@@ -79,27 +76,25 @@ public void removeAgent(int id)
...
 
-        private Agent(URI uri, boolean isCoordinator, boolean isWorker)
+        private Agent(URI uri, boolean isCoordinator, boolean isWorker, UUID id)
         {
             this.uri = requireNonNull(uri, "uri is null");
             this.isCoordinator = isCoordinator;
             this.isWorker = isWorker;
-
-            this.id = count.incrementAndGet();
+            this.id = requireNonNull(id, "null agent id");
```
Comment (ShanBai6) [inline presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/NodeSet.java]: put the ones with requireNonNull together.

File: pom.xml (hunk 1 above)
Comment (jirassimok) [inline pom.xml]: We'll remove it once we're not using it anymore; the note is just there as a reminder.

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/APIFileHandler.java (hunk 2 above)
Comment (jirassimok) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/APIFileHandler.java]: Done

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoUninstaller.java (hunk 9 above)
Comment (jirassimok) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoUninstaller.java]: Looking at the function, the uninstall command runs in a different conditional branch; the only thing after this line is another log entry.

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/api/ConfigAPI.java (hunk 10 above)
Comment (jirassimok) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/api/ConfigAPI.java]: Done.

File: pom.xml (hunk 3 above)
Comment (jirassimok) [inline pom.xml]: No, I'll make this change tomorrow.

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentServer.java (hunk 4 above)
Comment (jirassimok) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentServer.java]: Done.

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentServer.java (hunk 5 above)
Comment (jirassimok) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentServer.java]: As long as the servers log to different files, this is unambiguous.

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/CommandExecutor.java (hunk 7 above)
Comment (jirassimok) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/CommandExecutor.java]: Adding the errors will be done in a separate commit/pull request.

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/LogsHandler.java (hunk 8 above)
Comment (jirassimok) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/LogsHandler.java]: It doesn't have a constant representing "all levels", which makes it less useful for our purposes.

File: presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/ControllerServerModule.java (hunk 12 above)
Comment (jirassimok) [inline presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/ControllerServerModule.java]: Oh, right, we have to inject that. I totally forgot. I'll do that tomorrow.

File: presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/NodeSet.java (hunk 13 above)
Comment (jirassimok) [inline presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/NodeSet.java]: I put them in parameter order.

We can make that sort of change when we switch to a map-based NodeSet (that's the next thing I'm planning after finishing the switch to airlift.)

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/APIFileHandler.java [hunk 14]
```diff
@@ -60,15 +59,15 @@ public Response getFile(String path)
     {
         try {
             String fileContent = AgentFileUtils.getFile(Paths.get(baseDir.toString(), path));
-            LOGGER.debug("Successfully retrieved the content of file {}", path);
+            LOGGER.debug("Successfully retrieved contents of file '%s'", path);
```
Comment (sudarsanm94) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/APIFileHandler.java]: _the_ contents of _the_ file

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/APIFileHandler.java [hunk 15]
```diff
@@ -60,15 +59,15 @@ public Response getFile(String path)
...
             return Response.status(Status.OK).entity(fileContent).build();
         }
         catch (FileNotFoundException e) {
-            LOGGER.error("File {} not found", path, e);
+            LOGGER.error(e, "File '%s' not found", path);
             return Response.status(Status.NOT_FOUND).build();
         }
         catch (IOException e) {
-            LOGGER.error("Failed to process file {}", path, e);
+            LOGGER.error(e, "Failed to process file '%s'", path);
```
Comment (sudarsanm94) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/APIFileHandler.java]: _the_ file

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentServer.java [hunk 16]
```diff
@@ -13,49 +13,47 @@
...
-                .setPort(PORT)
-                .registerComponent(ConfigAPI.class)
-                .registerComponent(ConnectorsAPI.class)
-                .registerComponent(LogsAPI.class)
-                .registerComponent(PackageAPI.class)
-                .build();
+        // TODO: Replace placeholder properties with proper configuration
+        System.setProperty("node.environment", "test");
+        System.setProperty("http-server.http.port", "8081");
+        System.setProperty("log.levels-file", "etc/log.properties");
```
Comment (sudarsanm94) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentServer.java]: I this this path should be `/etc/presto/log.properties` for RPM installation.

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentServer.java (hunk 5 above)
Comment (sudarsanm94) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentServer.java]: But when we are looking at logs on the terminal. It would be more helpful if the error message is `Error running Agent server`

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentServerModule.java [hunk 17]
```diff
@@ -0,0 +1,38 @@
...
+{
+    @Override
+    protected void setup(Binder binder)
+    {
+        binder.disableCircularProxies();
+
+        jaxrsBinder(binder).bind(ConfigAPI.class);
+        jaxrsBinder(binder).bind(ConnectorsAPI.class);
+        jaxrsBinder(binder).bind(LogsAPI.class);
+        jaxrsBinder(binder).bind(PackageAPI.class);
```
Comment (sudarsanm94) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentServerModule.java]: missing `ControlAPI.class`

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoUninstaller.java (hunk 9 above)
Comment (sudarsanm94) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoUninstaller.java]: No after this warn line the actual uninstallation happens.

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentServer.java (hunk 16 above)
Comment (jirassimok) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentServer.java]: This is configuration for Presto Manager, not Presto.

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentServerModule.java (hunk 17 above)
Comment (jirassimok) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentServerModule.java]: Hasn't been rebased onto that commit yet.

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/APIFileHandler.java (hunk 14 above)
Comment (jirassimok) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/APIFileHandler.java]: I prefer to keep my error messages as simple as possible, and I prefer not to mix "grammar levels".
I could go for "Successfully retrieved the contents of the file at the path '%s'", but that would be pressing the verbosity a bit far, and doesn't carry any more information.

File: presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/ControllerServerModule.java (hunk 12 above)
Comment (jirassimok) [inline presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/ControllerServerModule.java]: Done.

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentServer.java (hunk 16 above)
Comment (jirassimok) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentServer.java]: This is the path for Presto Manager logs, not Presto logs.

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoRpmController.java [hunk 18]
```diff
@@ -40,20 +41,24 @@
...
-    private static final Logger LOGGER = LogManager.getLogger(PrestoRpmControlUtils.class);
-    private static final JerseyClient JERSEY_CLIENT = createClient();
+    private static final Logger LOGGER = LogManager.getLogger(PrestoRpmController.class);
     private static final String LAUNCHER_SCRIPT = "/usr/lib/presto/bin/launcher";
     private static final Path PRESTO_CONFIG_FILE = Paths.get("/etc/presto/config.properties");
     private static final Gson GSON = new Gson();
     private static final int SUBPROCESS_TIMEOUT = 120;
 
-    private PrestoRpmControlUtils() {}
+    private final Client CLIENT;
```
Comment (maciejgrzybek) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoRpmController.java]: It seems like broken convention: CAPITAL_CASE tends to be used for static constants only (`static final`) while it's not `static`. Why not just `client`?

File: presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/ControllerServer.java [hunk 19]
```diff
@@ -0,0 +1,43 @@
+// Temporary file to add directory to git.
```
Comment (maciejgrzybek) [inline presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/ControllerServer.java]: License is missing.

Comment (maciejgrzybek) [inline presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/ControllerServer.java]: BTW Why don't you have setup a license checker in Maven?

File: presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/ControllerServer.java [hunk 20]
```diff
@@ -0,0 +1,43 @@
...
+import io.airlift.discovery.client.DiscoveryModule;
+import io.airlift.event.client.HttpEventModule;
+import io.airlift.http.server.HttpServerModule;
+import io.airlift.jaxrs.JaxrsModule;
+import io.airlift.json.JsonModule;
+import io.airlift.node.NodeModule;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+class ControllerServer
```
Comment (maciejgrzybek) [inline presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/ControllerServer.java]: `public class`

File: presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/NodeSet.java [hunk 21]
```diff
@@ -28,14 +30,10 @@
...
-    private static NodeSet nodeSetInstance = new NodeSet();
-
-    public static Set<Agent> nodeSet = ConcurrentHashMap.newKeySet();
+    private Set<Agent> nodeSet = ConcurrentHashMap.newKeySet();
 
-    public static NodeSet getInstance()
-    {
-        return nodeSetInstance;
-    }
+    @Inject
```
Comment (maciejgrzybek) [inline presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/NodeSet.java]: You don't inject here anything.

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoUninstaller.java [hunk 22]
```diff
@@ -61,14 +60,14 @@ private static void uninstallUsingRpm(boolean checkDependencies, boolean ignoreE
...
-            LOGGER.warn("Package '{}' is not installed; Process exited with return value: {}", PRESTO_PACKAGE, checkPackageInstalled);
+            LOGGER.warn("Package '%s' is not installed; Process exited with return value: %s", PRESTO_PACKAGE, checkPackageInstalled);
             return;
         }
         if (executeCommand("service", "presto", "status") == 0) {
             if (!ignoreErrors) {
                 throw new PrestoManagerException("Presto is running");
             }
-            LOGGER.warn("Presto is running");
+            LOGGER.warn("Presto is running, but uninstall will be attempted; Presto will be unstoppable");
```
Comment (maciejgrzybek) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoUninstaller.java]: I like the "unstoppable" motive. ;)

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/api/ConfigAPI.java [hunk 23]
```diff
@@ -44,7 +42,6 @@
 public final class ConfigAPI
 {
     private static final APIFileHandler apiFileHandler = new APIFileHandler(Paths.get("/etc/presto"));
-    private static final Logger LOGGER = LogManager.getLogger(ConfigAPI.class);
```
Comment (maciejgrzybek) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/api/ConfigAPI.java]: Why did you decide to remove logger from this class? Is the logging of the requests handled in `ApiFileHandler`? If so, I'd extract this change into a separate commit because it's not related to migration to Airlift.

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/api/ConnectorsAPI.java [hunk 24]
```diff
@@ -43,7 +41,6 @@
 public final class ConnectorsAPI
 {
     private static final APIFileHandler apiFileHandler = new APIFileHandler(Paths.get("/etc/presto/catalog"));
-    private static final Logger LOGGER = LogManager.getLogger(ConnectorsAPI.class);
```
Comment (maciejgrzybek) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/api/ConnectorsAPI.java]: ditto

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/api/ControlAPI.java [hunk 25]
```diff
@@ -41,7 +40,7 @@
 @Singleton
 public class ControlAPI
 {
-    private static final Logger LOGGER = LogManager.getLogger(ControlAPI.class);
+    private static final Logger LOGGER = Logger.get(ControlAPI.class);
```
Comment (maciejgrzybek) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/api/ControlAPI.java]: Why you need it given you removed all usages of logger?
Also, ditto on removal in separate commit.

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/api/LogsAPI.java [hunk 26]
```diff
@@ -47,8 +45,6 @@
         date = "2017-06-23T09:53:13.549-04:00")
 public class LogsAPI
 {
-    private static final Logger LOG = LogManager.getLogger(LogsAPI.class);
```
Comment (maciejgrzybek) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/api/LogsAPI.java]: ditto

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/api/PackageAPI.java [hunk 27]
```diff
@@ -55,7 +55,8 @@
 // TODO: Add configuration to get package type from PM config file
 public final class PackageAPI
 {
-    private static final Logger LOGGER = LogManager.getLogger(PackageAPI.class);
+    private static final Logger LOGGER = Logger.get(PackageAPI.class);
+    private static PackageType installedPackageType;
```
Comment (maciejgrzybek) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/api/PackageAPI.java]: It's another unrelated to logger migration change.

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentServer.java [hunk 28]
```diff
@@ -53,7 +53,7 @@ public static void main(String[] args)
             injector.getInstance(Announcer.class).start();
         }
         catch (Exception e) {
-            LOG.error(e, "Error running application");
+            LOG.error(e, "Error starting server");
```
Comment (maciejgrzybek) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentServer.java]: Unrelated change.

File: presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/api/ControllerConnectorAPI.java [hunk 29]
```diff
@@ -58,15 +59,15 @@ public ControllerConnectorAPI(
         return ApiRequester.builder(forwardingClient, clazz);
     }
 
-    private Response forwardRequest(String scope, ApiRequester requester, Collection<Integer> nodeId)
+    private Response forwardRequest(String scope, ApiRequester requester, Collection<UUID> nodeId)
```
Comment (maciejgrzybek) [inline presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/api/ControllerConnectorAPI.java]: Don't you need to change the underlying class too?

File: presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/ControllerServer.java (hunk 19 above)
Comment (jirassimok) [inline presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/ControllerServer.java]: Fixed.
I think we have a TODO in our POM to set that up... How do we do it?

File: presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/ControllerServer.java (hunk 20 above)
Comment (jirassimok) [inline presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/ControllerServer.java]: Fixed

File: presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/NodeSet.java (hunk 21 above)
Comment (jirassimok) [inline presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/NodeSet.java]: Oops.
Then the annotation has no effect at all here. I'd rather leave it to minimize conflicts with the `discovery` branch, where something _does_ get injected.

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/api/ConfigAPI.java (hunk 23 above)
Comment (jirassimok) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/api/ConfigAPI.java]: All HTTP requests are logged by Airlift itself.
though I've noticed there is a bit less detail... We might want to add these back to all of the API classes later.
Better yet, we might be able to use Jersey's `LoggingFeature` to do it for us.

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/api/PackageAPI.java (hunk 27 above)
Comment (jirassimok) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/api/PackageAPI.java]: That was a mis-merge. It's fixed in a later commit.

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentServer.java (hunk 28 above)
Comment (jirassimok) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentServer.java]: This should be in the last commit of the PR ("changes from code review"), but I can't rewrite the history any more besides that last commit.

File: presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/api/ControllerConnectorAPI.java (hunk 29 above)
Comment (jirassimok) [inline presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/api/ControllerConnectorAPI.java]: I did.

File: presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoRpmController.java (hunk 18 above)
Comment (jirassimok) [inline presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoRpmController.java]: That's fixed somewhere in the PR.

**Comment by jirassimok** [comment]
All comments addressed.
